WSL/disk-space.md

```diff
@@ -0,0 +1,217 @@
+
+## How to repair a VHD mounting error
+
+If you encounter an error related to "mounting the distribution disk", this could be due to a sudden shutdown or power outage and may result in the Linux distribution VHD being switched to *read-only* to avoid data loss. You can repair and restore the distribution using the `e2fsck` Linux command by following the steps below.
+
+### Use the lsblk command to identify the block device name
+
+When WSL 2 installs a Linux distribution, it is mounting the distribution as a Virtual Hard Disk (VHD) with it's own file system. Linux refers to these hard drives as "block devices" and you can view information about them by using the `lsblk` command.
+
+To find the names of the block devices currently being used by WSL 2, open your distribution and enter the command: `lsblk`. (Or open PowerShell and enter the command: `wsl.exe lsblk`.) The output will look something like this:
+
+```bash
+NAME MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
+sda    8:0    0 363.1M  1 disk
+sdb    8:16   0     8G  0 disk [SWAP]
+sdc    8:32   0   1.5T  0 disk
+sdd    8:48   0     1T  0 disk /mnt/wslg/distro
+```
+
+Information about the block device includes:
+
+- **NAME**: The name assigned to the device will be sd[a-z], referring to the SCSI Disk with a letter designation for each disk being used. `sda` is always the system distribution.
+- **MAJ:MIN**: Represents numbers used by the Linux kernel to internally identify the devices with the first number representing the device type (8 is used for Small Computer System Interface/SCSI disks, an alternative to USB).
+- **RM**: Let's us know if the device is removable (1) or not (0).
+- **SIZE**: lets us know how much space the device is using.
+- **RO**: Let's us know if the device is read-only (1) or not (0).
+- **TYPE**: Refers to the device type (disk in this case, could also be something like lvm for a logical volume device).
+- **MOUNTPOINTS**: Refers to the current directory on the files system where the block device is located (SWAP is for preconfigured inactive memory so no mountpoint).
+
+### Read-only fallback error
+
+If WSL encounters an error while installing (mounting) a Linux distribution, it may mount the distribution as read-only (RO = 1) as a fallback. If that happens, the distribution may display the following error during startup:
+
+```powershell
+An error occurred mounting the distribution disk, it was mounted read-only as a fallback.
+```
+
+When a distribution is started as read-only, any attempts to write to the filesystem will fail with an error like this:
+
+```bash
+$ touch file
+touch: cannot touch 'file': Read-only file system
+```
+
+To repair a disk mount error in WSL, and restore it back to a usable / writeable state again, you can use the `wsl --mount` command to re-mount the disk with the following steps:
```

Comment:
Let's use `wsl.exe` everywhere as the original pull request did (this protects from powershell aliases / potential conflicting commands)

Reply:
Updated.